File: 백준/Silver/10845. 큐/큐.java
import java.io.*;
import java.util.*;

public class Main {

    static ArrayDeque<String> dq = new ArrayDeque<>();
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());

        for(int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();
            queueCmd(cmd);
        }
        br.close();
        System.out.println(sb.toString());
    }

    static void queueCmd(String word) {
        switch (word) {
            case "push":
                dq.add(st.nextToken());
                break;

            case "pop":
                if (dq.isEmpty()) {
                    sb.append(-1);
                } else {
                    sb.append(dq.remove());
                }
                sb.append("\n");
                break;

            case "size":
                sb.append(dq.size()).append("\n");
                break;

            case "empty":
                if (dq.isEmpty()) {
                    sb.append(1);
                } else {
                    sb.append(0);
                }
                sb.append("\n");
                break;

            case "front":
                if (dq.isEmpty()) {
                    sb.append(-1);
                } else {
                    sb.append(dq.peek());
                }
                sb.append("\n");
                break;

            case "back":
                if (dq.isEmpty()) {
                    sb.append(-1);
                } else {
                    sb.append(dq.peekLast());
                }
                sb.append("\n");
                break;
        }
    }
}

File: 백준/Silver/10845. 큐/큐.kt
import java.util.*
import java.io.*
import kotlin.collections.ArrayDeque

fun main() = with(BufferedReader(InputStreamReader(System.`in`))) {

    val N = readLine().toInt()
    val queue = ArrayDeque<Int>()
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    repeat(N) {
        val st = StringTokenizer(readLine())
        when(st.nextToken()) {
            "push" -> queue.addLast(st.nextToken().toInt())
            "pop" -> if(queue.isEmpty()) bw.write("-1\n") else bw.write("${queue.removeFirst()}\n")
            "size" -> bw.write("${queue.size}\n")
            "empty" -> if(queue.isEmpty()) bw.write("1\n") else bw.write("0\n")
            "front" -> if(queue.isEmpty()) bw.write("-1\n") else bw.write("${queue.first()}\n")
            "back" -> if(queue.isEmpty()) bw.write("-1\n") else bw.write("${queue.last()}\n")
        }
    }
    close()
    bw.flush()
    bw.close()
}